Shopify/hydrogen
# Fix info banners during init

### WHY are these changes introduced?

Fixes https://github.com/Shopify/hydrogen-internal/issues/44



### WHAT is this pull request doing?

Modify's the init/create flow to only show the banner about inventory when template chosen is the demo-store.



### HOW to test your changes?

Run init or create locally and choose the hello-world template and verify that no inventory banner is displayed.

#### Post-merge steps



#### Checklist
